File: pytools/data/data.code.golang.server/ack_scene_enter.go
package proto

import (
	"packet"
)

type AckSceneEnter struct {
	player                   *MsgScenePlayer
}

func (this *AckSceneEnter) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteBytes(this.player.Encode())

	return pack.Encode(P_ACK_SCENE_ENTER)
}

func (this *AckSceneEnter) SetPlayer(player *MsgScenePlayer) {
	this.player = player
}

File: pytools/data/data.code.cpp.server/AckSceneEnter.cpp
#ifndef _ACK_SCENE_ENTER_
#define _ACK_SCENE_ENTER_

#include <list>

#include <pb_type.hpp>
#include <Packet.hpp>

#include <Msg.hpp>

#include <MsgScenePlayer.cpp>


class AckSceneEnter
{
private:
	MsgScenePlayer* m_player;


public:
	Packet Encode()
	{
		Packet packet;
		U32 player_buff_len = 0;
		BYTE* player_buff = m_player->GetBytes(&player_buff_len);
		packet.WriteBytes(player_buff, player_buff_len);
		packet.Encode(Msg::P_ACK_SCENE_ENTER);
		return packet;
	}


	void SetPlayer(MsgScenePlayer* player)
	{
		m_player = player;
	}

};

#endif
